source/agora/common/BlockStorage.d

```diff
@@ -61,8 +61,10 @@ private struct HashPosition
     size_t              position;
 }
 
-alias IndexHeight = RedBlackTree!(HeightPosition, "(a.height < b.height)");
-alias IndexHash = RedBlackTree!(HashPosition, "(a.hash < b.hash)");
+/// Type of RBTree used for height indexing
+private alias IndexHeight = RedBlackTree!(HeightPosition, "(a.height < b.height)");
```

Comment:
Could we use lambdas instead?

```D
private alias IndexHeight = RedBlackTree!(HeightPosition, (a, b) => a.height < b.height);
```

It avoids having to mixin strings inside of Phobos, which speeds up compilation and also makes error diagnostics better.

Reply:
Well this commit is just about making it private. I actually want to re-explore the use of RBTree here, especially for height.